src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -110,12 +110,33 @@ internal partial class Binder
 
             if (conversion.Kind == ConversionKind.SwitchExpression)
             {
-                return ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, destination, conversionIfTargetTyped: conversion, diagnostics);
+
+                var convertedSwitch = ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, destination, conversionIfTargetTyped: conversion, diagnostics);
+                return new BoundConversion(
+                    syntax,
+                    convertedSwitch,
+                    conversion,
+                    CheckOverflowAtRuntime,
+                    explicitCastInCode: isCast && !wasCompilerGenerated,
```

Comment:
What happens when `&& !wasCompiledGenerated` is removed from this argument?

Reply:
We violate the meaning of explicitCastInCode :). That property specifically reflects whether the user typed `(Type)`, and if the compiler generated the cast it wasn't typed by the user.

---
In reply to: [592881226](https://github.com/dotnet/roslyn/pull/51791#discussion_r592881226) [](ancestors = 592881226)

Reply:
I'm slightly concerned about this, because synthesized code can contain explicit conversions, and NullableWalker can take such conversions and try to re-classify them, passing its `ExplicitCastInCode` value to `ClassifyConversion`'s `forCast` argument (indirectly). Then Conversions will say that the conversion doesn't exist. Wrong information could be propagated out, or unwanted diagnostics given. See #51631.

Basically, it's a rather small corner case, but I think if `ExplicitCastInCode` *requires* that a cast is present in user syntax, we may want a follow up item to make sure the NullableWalker avoids using it as an argument to `NullableWalker.GenerateConversion`, and to make sure that synthesized code never passes a 'true' value for this flag. Possibly with `Debug.Assert(!(WasCompilerGenerated && ExplicitCastInCode))` when creating a BoundConversion or something like that.

Reply:
I also feel like there is likely something I'm missing about the general purpose of `ExplicitCastInCode`, something that is not addressed by just `Conversion.IsExplicit`.

Reply:
> something that is not addressed by just Conversion.IsExplicit

Whether or not a conversion is explicit is unrelated to whether it actually existed in code. You can explicitly specify an implicit conversion, or the only conversion that exists could be explicit and not be specified in source (which would lead to an error). I think any concerns here are beyond the scope of the changes in this method, feel free to file a bug for follow up later. Note this is how we calculate `ExplicitCastInCode` in all cases in this method, it's not something I'm coming up with here.

---
In reply to: [594763328](https://github.com/dotnet/roslyn/pull/51791#discussion_r594763328) [](ancestors = 594763328)

Reply:
Have filed #51896, thanks.